docs/technical_reference/unit_models/energy_models/thermal_energy_storage.rst

```diff
@@ -0,0 +1,92 @@
+Thermal Energy Storage (TES)
+====================================================
+
+This is thermal energy storage (TES) model assumes the tank is at a uniform temperature (similar to a continuous stirred tank) and supports steady-state only.
+
+Model Structure
+---------------
+
+This TES model consists of 4 StateBlocks (as 4 Ports in parenthesis below). Two state blocks connect
+to the the external heat exchanger which adds heat to the TES and two ports connect to the process side
+and provide heat to the treatment process.
+
+* Heat exchanger inlet (tes_hx_inlet)
+* Heat exchanger outlet (tes_hx_outlet)
+* Process inlet (tes_process_inlet)
+* Process outlet (tes_process_outlet)
+
+Sets
+----
+
+.. csv-table::
+   :header: "Description", "Symbol", "Indices"
+
+   "Time", ":math:`t`", "[0]"
+   "Phases", ":math:`p`", "['Liq', 'Vap']"
+   "Components", ":math:`j`", "['H2O']"
+
+Degrees of Freedom/Variables
+------------------
+
+The TES model has 4 degrees of freedom that should be fixed for the unit to be fully specified
+in addition to the state variables at the inlet and outlet.
+Typically the following variables define the input and output heat exchanger and process ports listed above. 
+
+.. csv-table::
+   :header: "Variables", "Variable name", "Symbol", "Valid range", "Unit"
+
+   "Temperature", "``temperature``", ":math:`T_{f}`", "298.15 - 363.15", ":math:`\text{K}`"
+   "Pressure", "``pressure``", ":math:`P`", "", ":math:`Pa`"
+   "Mass flow rate of vapor phase", "``flow_mass_phase_comp[0,'Vap','H2O']``", ":math:`m_{v}`", "", ":math:`kg/s`"
+   "Mass flow rate of liquid phase", "``flow_mass_phase_comp[0,'Liq','H2O']``", ":math:`m_{l}`", "", ":math:`kg/s`"
+   
+The following variables should also be fixed. An initial temperature is assigned to the outlet stream at the heat exhanger and process loop.
+
+.. csv-table::
+   :header: "Variables", "Variable name", "Symbol", "Valid range", "Unit"
+
+   "Initial temperature", "``tes_initial_temp``", ":math:`T_{0}`", "298.15 - 363.15", ":math:`\text{K}`"
+   "Time step", "``dt``", ":math:`dt`", "", ":math:`h`"
+   "Hours of storage", "``hours_storage``", ":math:`h_{storage}`", "0-24", ":math:`h`"
+   "Heat load", "``heat_load``", ":math:`heat_{load}`", "", ":math:`MW`"
+   
+
+Parameters
+----------
+
+The following parameters are used as default values and are mutable. 
+
+.. csv-table::
+   :header: "Description", "Parameter Name", "Symbol", "Value", "Units"
+
+   "Salt packing density", "``salt_packing_density``", ":math:`\rho_{packing,salt}`", "1", ":math:`kg/s`"
```

Comment:
Is this unit supposed to represent molten salt energy storage? That's my interpretation from seeing mentions of "salt packing density." 

If so, I thought molten salt energy storage was applicable to high-temp CSP (e.g., power tower, parabolic trough) but unsure about applicability to FPCs.

Reply:
@MuktaHardikar is going to modify the TES model to be for water rather than salt in a separate PR. The TES docs in this PR will be for water.